pkg/crc/machine/machine.go

```diff
@@ -128,6 +128,19 @@ func (client *client) updateVMConfig(startConfig StartConfig, api libmachine.API
 		return err
 	}
 
+	/* Disk size */
+	if startConfig.DiskSize != 0 {
+		if err := setDiskSize(host, startConfig.DiskSize); err != nil {
+			logging.Debugf("Failed to update CRC disk configuration: %v", err)
+			if err != drivers.ErrNotImplemented {
+				return err
```

Comment:
I guess this is where macOS user will currently fall ?

Reply:
Same my concern, `startConfig.DiskSize` will have the default DiskSize, shouldn't we compare `startConfig.DiskSize != constants.DefaultDiskSize` ?

Reply:
This is checking for `UpdateConfigRaw` failures, at the moment the hyperkit driver only ignores disk sizes, so it won't return an error, it needs an update in order to return an error. This is one part where `UpdateConfigRaw` could be improved, return an error if it was called but did not  modify any fields.

Reply:
> Same my concern, startConfig.DiskSize will have the default DiskSize, shouldn't we compare startConfig.DiskSize != constants.DefaultDiskSize ?

I don't understand what you are asking :( However, I did not want the 'disk-size' config to default to `constants.DefaultDiskSize` as this would cause unexpected behaviour when doing
`crc start --disk-size 50; crc stop; crc start`.
It would try to resize the disk back to `constants.DefaultDiskSize`, but the libvirt driver errors out when trying to do that. So it's better to keep the old size, and `0` is used as a means to indicate "no value was set"

Reply:
> This is one part where UpdateConfigRaw could be improved, return an error if it was called but did not modify any fields.

The `machine-driver-hyperkit` implementation may be something like this: (untested)
```
From c83e1a0c18f5df18940e686dd51737bd93fe67d1 Mon Sep 17 00:00:00 2001
From: Christophe Fergeau <cfergeau@redhat.com>
Date: Wed, 21 Oct 2020 20:36:26 +0200
Subject: [PATCH] Report an error when trying to change unsupported field

Only memory size and number of CPUS should be changed with
`UpdateConfigRaw`. If neither of these changed, the call
to UpdateConfigRaw is unsupported.
---
 pkg/hyperkit/driver.go | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/pkg/hyperkit/driver.go b/pkg/hyperkit/driver.go
index 084b54e..c8bbd90 100644
--- a/pkg/hyperkit/driver.go
+++ b/pkg/hyperkit/driver.go
@@ -416,6 +416,13 @@ func (d *Driver) UpdateConfigRaw(rawConfig []byte) error {
 		return err
 	}
 
+	if newDriver.Memory == d.Memory && newDriver.CPU == d.CPU {
+		/* For now only changing memory and CPU is supported/tested.
+		 * If none of these changed, we might be trying to change another
+		 * value, which is may or may not work, return ErrNotImplemented for now
+		 */
+		return drivers.ErrNotImplemented
+	}
 	*d = newDriver
 
 	return nil
-- 
2.28.0
```

Reply:
I tried on macOS and the flag is just discarded. I guess it's fine for a first release.

Reply:
https://github.com/code-ready/machine-driver-hyperkit/pull/21 should improve this, together with the commit I added to this PR to show a warning when `ErrNotImplemented` is returned.

Reply:
code-ready/machine-driver-hyperkit#21 is not for disk size. Should we add it ?